src/app/home/problem-list/problem-table/problem-table.component.ts

```diff
@@ -24,6 +23,7 @@ export enum ProblemStatus {
 export class ProblemTableComponent implements OnInit {
   @ViewChild(MatSort) sort: MatSort;
   @ViewChild(MatPaginator) paginator: MatPaginator;
+  @Input() userInfo: IUserInfo;
```

Comment:
I'm curious, what is rationale for making this an `@Input` and not injecting it directly here?